src/pages/AddServers/DeployAddServers.js

```diff
@@ -0,0 +1,253 @@
+import { translate } from '../../localization/localize.js';
+import {
+  STATUS, WIPE_DISKS_PLAYBOOK, ARDANA_GEN_HOSTS_FILE_PLAYBOOK,
+  SITE_PLAYBOOK, MONASCA_DEPLOY_PLAYBOOK
+} from '../../utils/constants.js';
+import { fetchJson } from '../../utils/RestUtils.js';
+
+
+//ansible-playbook -i hosts/verb_hosts wipe_disks.yml --limit <hostname,hostname,hostname>
+//ansible-playbook -i hosts/verb_hosts ardana-gen-hosts-file.yml"
+//ansible-playbook -i hosts/verb_hosts site.yml --limit <hostname, hostname, hostname>
+//ansible-playbook -i hosts/verb_hosts monasca-deploy.yml --tags "active_ping_checks"
+
+let PLAYBOOK_POSSIBLE_STEPS = [
+  {
+    name: WIPE_DISKS_PLAYBOOK,
+    label: translate('server.deploy.progress.wipe-disks'),
+    playbooks: [WIPE_DISKS_PLAYBOOK + '.yml'],
+    payload: {limit: {}}
+  },
+  {
+    name: ARDANA_GEN_HOSTS_FILE_PLAYBOOK,
+    label: translate('server.deploy.progress.gen-hosts-file'),
+    playbooks: [ARDANA_GEN_HOSTS_FILE_PLAYBOOK + '.yml']
+  },
+  {
+    name: SITE_PLAYBOOK,
+    label: translate('server.deploy.progress.addserver.deploy'),
+    playbooks: [SITE_PLAYBOOK + '.yml'],
+    payload: {limit: {}}
+  },
+  {
+    name: MONASCA_DEPLOY_PLAYBOOK,
+    label: translate('server.deploy.progress.update-monasca'),
+    playbooks: [MONASCA_DEPLOY_PLAYBOOK + '.yml'],
+    payload: {tags: 'active_ping_checks'}
+  }
+];
+
+class DeployAddServers extends BaseUpdateWizardPage {
+
+  constructor(props) {
+    super(props);
+
+    this.playbooks = [];
+    this.steps = [];
+
+    this.state = {
+      overallStatus: STATUS.UNKNOWN, // overall status of entire playbook and commit
+      processErrorBanner: '',
+      newHosts: this.getNewHosts(),
+      // loading errors from wizard model or progress loading
+      wizardLoadingErrors: this.props.wizardLoadingErrors,
+      // loading indicator from wizard
+      wizardLoading: this.props.wizardLoading,
+      // this loading indicator
+      loading: false,
+      // warning message
+      warningMessage: undefined
+    };
+  }
+
+  getNewHosts = () => {
+    return (
+      this.props.operationProps && this.props.operationProps.newHosts ?
+        this.props.operationProps.newHosts : []
+    );
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.setState({
+      wizardLoadingErrors: newProps.wizardLoadingErrors,
+      wizardLoading: newProps.wizardLoading,
+      newHosts: newProps.operationProps.newHosts
+    });
+  }
+
+  componentDidMount() {
+    // go get hostnames, if there are no recorded found
+    // this will take some time
+    // will request with no-cache
+    if(!this.props.operationProps.newHosts) {
+      this.setState({loading: true});
+      fetchJson(
+        '/api/v1/clm/model/cp_internal/CloudModel.yaml', undefined, true, true
+      )
+        .then((cloudModel) => {
+          let newHosts = this.getAddedComputeHosts(cloudModel);
+          let cleanedHosts = newHosts.filter(host => host['hostname'] !== undefined);
```

Comment:
as we discussed this is a lot of extra logic to cover for a bug... no reason to remove it for now, maybe add a comment about the bugzilla ID of the bug?

Reply:
ok ..will file a bug for ansible and update with comments